Feature - integration with jss expand

typical000 opened the pull request:
1. Migrated tests to mocha (@kof don't forget to add after merging PR default unit to test suite in main jss repository)
2. Added cross-end ( https://github.com/cssinjs/jss/issues/323 )
3. Added tests to cover situations of recursive tree passing (included jss-nested for tests)
4. Added tests to cover situation described in https://github.com/cssinjs/jss-expand/issues/12 e.g:
``````````````js
const sheet = jss.createStyleSheet({
  container: {
    top: 20, // This is 'positioning' property. Will be taken value from 'top' property in defaultUnit.js
    padding: {
      top: 5, // This is padding partial property. Property will be compared with 'padding-top' from defaultUnit.js
      left: 10,
      bottom: 5
    }
  }
})
``````````````

NOTE: 
1. After merge - add default-unit to test suite in main jss repo.
2. Need to add **jss-expand** as dev dependency to main jss repo as a solution for https://github.com/cssinjs/jss/issues/320

kof commented on history.md:
```diff
@@ -1,3 +1,8 @@
+## 4.0.1 / 2016-09-25
```
4.0.2

kof replied:
oh it is 5.0.0, you made, do we have incompatible changes?

kof commented on karma.conf.js:
```diff
@@ -0,0 +1,25 @@
+
+module.exports = (config) => {
+  config.set({
+    browsers: ['Chrome', 'Firefox', 'Safari'],
```
I think its ok to have just Chrome here, because there is nothing that can be cross-browser incosistent I think.

kof commented on src/index.js:
```diff
@@ -14,6 +14,13 @@ function iterate(prop, value, options) {
   switch (value.constructor) {
     case Object:
+      // If property is not a 'nesting' rule - compose parent value with children property names.
+      if (prop.indexOf('&')) {
```
indexOf returns -1 when there is no match.

kof replied:
why do we need this, according to the order https://github.com/cssinjs/jss-preset-default/blob/master/src/index.js jss-nested has been applied already.

typical000 replied:
> why do we need this, according to the order https://github.com/cssinjs/jss-preset-default/blob/master/src/index.js jss-nested has been applied already.

But not all people can use jss-preset-default. Maybe we will add in README.md in jss-default-unit some words about order?

So, remove this indexOf and add to readme about order

kof replied:
Definitely, if a user uses nested selectors, we expect he uses jss-nested plugin. And we expect it is in the right order.

typical000 replied:
> And we expect it is in the right order.
Ok. You're right. It's my mistake. I just encountered some problems in finding info about "Order does matter" https://github.com/cssinjs/jss/blob/master/docs/plugins.md :)

typical000 commented:
So, added punch of fixes:
1. Changed version to 5.0.0
2. Removed jss-nested from tests and dependencies (My mistake, forgot about 'order does matter')
3. Removed indexOf from code
4. Removed FF and Safari from karma.conf.js

kof commented on package.json:
```diff
@@ -1,7 +1,7 @@
   "name": "jss-default-unit",
   "description": "JSS plugin that adds default custom unit to numeric values where needed",
-  "version": "4.0.1",
+  "version": "5.0.0",
```
Still the question is why 5, major version gets incremented in semver when we have  a breaking change. Do we have one?

kof replied:
If no, it should be 4.1.0

typical000 replied:
Looked to other plugins version history. You're right. Migration to mocha is not so "breaking" change", fixes for jss-expand - too. There is no new features inside this pr.

typical000 commented:
Changed version

typical000 commented:
Forgot about fallbacks. This issue https://github.com/cssinjs/jss/issues/325 make me remember this. Need to add some test cases

typical000 commented:
Added fallbacks check and tests https://github.com/cssinjs/jss-default-unit/pull/9/commits/f397aa3b440347e6eb8d73241f58df9fc106108b

typical000 commented:
Hi. What about this pr?

kof commented:
Huh totally forgot about it.

kof commented:
merged
